### DISCUSSION TITLE
Repos incorrectly classified as Euphoria

### DISCUSSION THREAD

**Discussion by ghaberek** [github-linguist/linguist#6880] [Classification] [answered]
## Problem description

I got pinged on [this tweet](https://x.com/stardev_io/status/1800601319603831150) (which is otherwise spam) but I noticed some repos are indeed classified as Euphoria when they are not.

I contributed the additions for Euphoria (#5241). Is there anything I should be concerned about? Any adjustments I should make?

### Affected repositories

| Repo | Stars | Last modified | Detected language | Expected language |
| --- | ---: | ---: | --- | --- |
| [Elixir-Development/Phoenix-Tutorial](/Elixir-Development/Phoenix-Tutorial) | 18 | 2022-09-02 | 100% Euphoria | Elixir |
| [SanaurAsif/roc-x](/SanaurAsif/roc-x) | 8 | 2022-09-11 | 50.9% Euphoria | Python |
| [phoenixding/scdiff2](/phoenixding/scdiff2) | 6 | 2022-10-25 | 98.8% Euphoria | Python |
| [patel-aum/ROT](/patel-aum/ROT) | 4 | 2023-05-23 | 63.3% Euphoria | Shell |
| [lanl/GeologicMeshing](/lanl/GeologicMeshing) | 2 | 2022-08-04 | 96.9% Euphoria | Text |
| [muntorg/munt-raspbian-repository](/muntorg/munt-raspbian-repository) | 2 | 2022-07-03 | 45.2% Euphoria | Shell |

**Comment by lildude** [comment]
This will be due to the shared extensions which means the only way of improving things is via the heuristics (preferred) or by adding more samples to hopefully get the classifier to be more accurate.

There's nothing to be concerned about; Linguist doesn't attempt  to be perfect. If those repo owners are particularly concerned, they can implement an override. Given the last modification date, I doubt they're particularly active and thus not concerned. Any changes you make to the heuristics wouldn't reflect in those repos until they made another push anyway.

**Comment by lildude** [comment, marked as answer]
I've just checked and the heuristics of `.e` have since been improved so it's possible those repos will sort themselves out the next time they're pushed to.
